docs/CODE_GUIDELINES.md

```diff
@@ -341,28 +358,49 @@ int const * const someIntPointer;
 
 ### 5.4. Initialization
 
-Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor. Zero-initialize with `{}`.
+Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor to zero-initialize with `{}`.
 
 ✅ Good:
 ```cpp
-int x{};
-int* y{};
-CLog::Log("test: {} {}", x, y);
+int x{3};
+CLog::Log("test: {}", x); // variable used just after its declaration
 ```
 
 ❌ Bad:
 ```cpp
-int x; // used uninitialized
-int* y = nullptr; // default-initialization not using {}
-CLog::Log("test: {} {}", x, y);
+int x{3}; // variable not immediately used, if possible declare the variable close to the point of its use 
```

Comment:
the example is not related to the rule text at all, since you are making up a new rule here. while I think it's a reasonable guideline, it should be explicitly written down and not "hidden" here

Reply:
add a new chapter see `5.1 Declarations`